builtin/providers/azurerm/resource_arm_subnet.go

```diff
@@ -96,6 +96,15 @@ func resourceArmSubnetCreate(d *schema.ResourceData, meta interface{}) error {
 		properties.RouteTable = &network.RouteTable{
 			ID: &rtId,
 		}
+
+		routeTableName, err := parseRouteTableName(rtId)
```

Comment:
routeTableName is needed in Lock operation below, can't limit its scope to the if { } block.